conans/client/remote_manager.py

```diff
@@ -150,24 +157,29 @@ def get_package(self, pref, dest_folder, remote, output, recorder):
 
             duration = time.time() - t1
             log_package_download(pref, duration, remote, zipped_files)
-            unzip_and_get_files(zipped_files, dest_folder, PACKAGE_TGZ_NAME, output=self._output)
+            unzip_and_get_files(zipped_files, unzip_folder, PACKAGE_TGZ_NAME,
+                                output=self._output,
+                                remove_tgz=not bool(package_install_folder))
             # Issue #214 https://github.com/conan-io/conan/issues/214
-            touch_folder(dest_folder)
+            touch_folder(unzip_folder)
             if get_env("CONAN_READ_ONLY_CACHE", False):
-                make_read_only(dest_folder)
+                make_read_only(package_folder)
```

Comment:
Maybe make read-only by default? Maybe it would be bad for imports, as imports would be read-only too, and will fail if repeated imports.

Reply:
Not sure either... sounds problematic, I don't know how often people are using the read-only cache.